chainer/function_set.py

```diff
@@ -28,8 +31,20 @@ def __init__(self, **functions):
                 object as attributes.
 
         """
-        for name, func in six.iteritems(functions):
-            setattr(self, name, func)
+
+        for key, func in six.iteritems(functions):
+            if key == 'name':
```

Comment:
This is for naming ``FunctionSet`` contained by another ``FunctionSet``. Better to add name only to ``Function`` ignoring ``FunctionSet``?